pkg/interactive/interactive.go

```diff
@@ -312,3 +338,64 @@ func PrintHelp(help Help) error {
 	fmt.Fprint(terminal.NewAnsiStdout(os.Stdout), out)
 	return nil
 }
+
+func ProcessParamInteractive(addOnInstallationParam *cmv1.AddOnInstallationParameter, param *cmv1.AddOnParameter, input *Input) (error, string) {
+	// Set default value based on existing parameter, otherwise use parameter default
+	dflt := param.DefaultValue()
+	if addOnInstallationParam != nil {
+		dflt = addOnInstallationParam.Value()
+	}
+
+	// add a prompt to question name to indicate if the boolean param is required and check validation
+	if param.ValueType() == "boolean" && param.Validation() == "^true$" && param.Required() {
+		input.Question = fmt.Sprintf("%s (required)", param.Name())
+		input.Validators = []Validator{
+			RegExpBoolean(param.Validation()),
+		}
+	}
+
+	var err error
+	var value string
+	switch param.ValueType() {
+	case "boolean":
+		var boolVal bool
+		input.Default, _ = strconv.ParseBool(dflt)
+		boolVal, err = GetBool(*input)
+		if boolVal {
+			value = "true"
+		} else {
+			value = "false"
+		}
+	case "cidr":
+		var cidrVal net.IPNet
+		if dflt != "" {
+			_, defaultIDR, _ := net.ParseCIDR(dflt)
+			input.Default = *defaultIDR
+		}
+		cidrVal, err = GetIPNet(*input)
+		value = cidrVal.String()
+		if value == "<nil>" {
+			value = ""
+		}
+	case "number", "resource":
+		if input.LenOptions() > 0 {
+			value, err = GetOption(*input)
```

Comment:
if we provide options, shouldn't that take precedence over the value type check and make it redundant?

Reply:
I think there can be options in the case of a number of the different types? Is the suggestion here to move this above the case statement and ignore the case if we have options?

Reply:
yes, that's what i'm thinking - in case we have options we don't need to go through the case statement